### PR TITLE
lib-user: add Grommet, zooTheme, storybook, and testing setup

### PR DESCRIPTION
**Pull request by mcbouslog**
## Package
- lib-user

## Linked Issue and/or Talk Post
Iteration on #5424 .

## Describe your changes
- add eslint config
- add Grommet and zooTheme
- add storybook
- add testing setup

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What user actions should my reviewer step through to review this PR?
  1. run `yarn panic && yarn bootstrap:es6`
  2. from `./packages/lib-user`
  3. run `yarn dev` - confirm dev server runs at `https://local.zooniverse.org:8080/?users=[your staging user login]/stats`
      - note: I often get an error on initial dev server load, though not after page refresh. I'm investigating, but do not have a solution at the moment
  4. run `yarn eslint` - confirm eslint output
  5. run `yarn test` - confirm mocha tests output (more on this in comments)
  6. run `yarn storybook` - confirm storybook at http://localhost:6008/?path=/story/components-groupstats--default
- Are there plans for follow up PR’s to further fix this bug or develop this feature? yes, see linked GitHub Project FEM Stats Efforts

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

### CODE REVIEW

File: packages/lib-user/package.json [hunk 1]
```diff
@@ -24,30 +24,59 @@
     "build:cjs": "babel ./src/ --out-dir ./dist/cjs --copy-files --no-copy-ignored",
     "build:es6": "BABEL_ENV=es6 babel ./src/ --out-dir ./dist/esm --copy-files --no-copy-ignored",
     "dev": "BABEL_ENV=es6 webpack serve --config webpack.dev.js",
+    "eslint": "eslint ./src --fix | snazzy",
     "prepare": "yarn build",
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "BABEL_ENV=es6 mocha --loader=@node-loader/babel --config ./test/.mocharc.json \"./src/**/*.spec.js\"",
```
Comment (mcbouslog) [inline packages/lib-user/package.json:29]: While getting mocha to run tests in the lib-user ESM package, I ran into similar issues noted in #3779 . The script here works, but there might be additional issues as more/actual tests are written. It's been discussed in Slack of trying Jest or an alternative to mocha.

File: packages/lib-user/src/components/GroupStats/GroupStats.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 'use client'
 
-import { string } from 'prop-types'
+import PropTypes from 'prop-types'
```
Comment (mcbouslog) [inline packages/lib-user/src/components/GroupStats/GroupStats.js:3]: This change is related to getting the `test` script (mocha test runner) to output as expected. See [this related comment](https://github.com/zooniverse/front-end-monorepo/pull/5475/files#r1369287374).

File: packages/lib-user/dev/components/App/App.js [hunk 3]
```diff
@@ -98,27 +101,48 @@ function App ({
     }
   }
 
+  const mergedThemes = Object.assign({}, baseTheme, zooTheme)
```
Comment (goplayoutside3) [inline packages/lib-user/dev/components/App/App.js:104]: A couple notes here:
1. `Object.assign` is outdated syntax and
2. I don't think manual merging is necessary for the Grommet provider anymore. In the [Grommet theming wiki](https://github.com/grommet/grommet/wiki/Grommet-v2-theming-documentation#basics), the example shows a custom theme object passed to the Grommet provider and notes that customization will just merge with the base theme. In app-root's RootLayout component I simply passed the `zooTheme`, so I think that's a strategy to try here too!

https://github.com/zooniverse/front-end-monorepo/blob/04fe35fcfba11d410d8c5cbec893621bfd9f9485/packages/app-root/src/components/RootLayout.js#L29

File: packages/lib-user/package.json (hunk 1 above)
Comment (goplayoutside3) [reply packages/lib-user/package.json:29]: Noting here just for Github visibility that I think it's fine to leave out mocha setup in this PR if you'd like to experiment with other testing solutions in lib-user (time allowing). Or if easier to move ahead with mocha that works too!

**Review by goplayoutside3 — APPROVED** [review]
Looks good! I was able to run all of the commands listed in the 'how to review' section 👍

File: packages/lib-user/package.json (hunk 1 above)
Comment (mcbouslog) [reply packages/lib-user/package.json:29]: I think I'd like to leave mocha in for now, but I'm thinking I add one (small) actual component with story and tests (with mocha), import the component into app-root to confirm works as expected, then attempt to refactor mocha with jest. Always open to suggestion, but that's my tentative plan for now.

File: packages/lib-user/dev/components/App/App.js (hunk 3 above)
Comment (mcbouslog) [reply packages/lib-user/dev/components/App/App.js:104]: Nice, thanks for noting!

**Pull request merged**